ci-operator/step-registry/idp/openldap/idp-openldap-commands.sh

```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+# Check if any IDP is already configured
+function check_idp () {
+    # Fetch the current identityProviders configuration if any
+    current_idp_config=$(oc get oauth cluster -o jsonpath='{.spec.identityProviders}')
+
+    if [ -n "$current_idp_config" ] && [ "$current_idp_config" != "null" ]; then
+        echo "An IDP is already configured. Skipping addition of new IDP."
+        exit 0
+    fi
+}
+
+# Function to set the proxy, if applicable
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "Setting the proxy"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "No proxy setting."
+    fi
+}
+
+# Function to set users if required
+function set_users() {
+    if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+        export KUBECONFIG=${SHARED_DIR}/kubeconfig
+    fi
+    users=""
```

Comment:
Seems what it does is not what the function name intends, i.e. to set users. It only initializes users="". So this misleads reader.

Reply:
added outside `set_users` function